04_numeronym/davidbroughsmyth/main_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := strconv.Quote("[a11y K8s abc]\n")
+	actual := strconv.Quote(buf.String())
+	assert.Equal(t, expected, actual, "Unexpected output from main()")
+}
+
+var tests = map[string]struct {
+	input, want []string
+}{
+	"empty":        {input: []string{}, want: []string{}},
+	"SpecialChars": {input: []string{"@@!##%&&"}, want: []string{"@6&"}},
+	"Alphanumeric": {input: []string{"8abcABC5Z"}, want: []string{"87Z"}},
+	"StarTrek":     {input: []string{"Beam", "me", "up", "Scotty"}, want: []string{"B2m", "me", "up", "S4y"}},
+	"MaryPoppins": {
+		input: []string{"It's", "supercalifragilisticexpialidocious"},
+		want:  []string{"I2s", "s32s"},
+	},
+}
```

Comment:
You should have a test with string of three runes, as that is a boundary condition. This ensures the function is not trying to shorten strings of 3 runes and that the condition for that case is implemented properly.

Reply:
I've added tests to handle the three rune boundary condition.